android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -633,6 +634,26 @@ class QuestionnaireViewModelTest : RobolectricTest() {
     Assert.assertEquals(questionnaireConfig.id, questionnaire?.id?.extractLogicalIdUuid())
   }
 
+  @Test
+  fun testRetrieveQuestionnaireShouldReturnValidQuestionnaireFromCache() = runBlocking {
+    coEvery { fhirEngine.get(ResourceType.Questionnaire, questionnaireConfig.id) } returns
+      samplePatientRegisterQuestionnaire
+
+    ContentCache.saveResource(questionnaireConfig.id, samplePatientRegisterQuestionnaire)
+
+    val questionnaire =
+      questionnaireViewModel.retrieveQuestionnaire(
+        questionnaireConfig = questionnaireConfig,
+      )
+
+    Assert.assertEquals(
+      samplePatientRegisterQuestionnaire,
+      ContentCache.getResource(ResourceType.Questionnaire.name + "/" + questionnaireConfig.id),
+    )
+    Assert.assertNotNull(questionnaire)
+    Assert.assertEquals(questionnaireConfig.id, questionnaire?.id?.extractLogicalIdUuid())
```

Comment:
Lets add a verify assertion here to verify that the 
`defaultRepository.loadResource<Questionnaire>(questionnaireConfig.id)` method was not called (i.e. was called zero times)

Reply:
And we can probably split this to 2 unit tests to get better coverage for the changes in this class.
- Test case 1 to test for when the questionnaire is Found in the cache
- Test case 2 to test for when the questionnaire is Not Found in the cache

So basically use the:
 
```kotlin
val questionnaire =
      questionnaireViewModel.retrieveQuestionnaire(
        questionnaireConfig = questionnaireConfig,
      ) 
 ```
 
method as the focus of the test case and leave out the ContentCache.save invocations since those are already tested in the `ContentCacheTest` class